src/js/service/collectElementService.js

```diff
@@ -218,6 +223,33 @@ collectElementService.doCollectElementActions = async function (action) {
                 })
             );
             break;
+        case GridActionCollectElement.COLLECT_ACTION_TOGGLE_TEXT_ROTATION:
+            // Find the collect element in the registered collect elements
+            if (registeredCollectElements && registeredCollectElements.length > 0) {
+                let collectElement = registeredCollectElements[0]; // Get the first (and usually only) collect element
+
+                // Toggle the rotation state for the collect element
+                collectElement.isTextRotated = !collectElement.isTextRotated;
+
+                // Save the grid to persist the change
+                // We need to get the grid ID and save it properly
+                let gridId = collectElement.gridId || (stateService.getCurrentGridId && stateService.getCurrentGridId());
```

Comment:
I think saving the property here is not really needed:
1. it seems not to work - if I toggle and then refreh the page/reload the app - the toggle state is again the initial one
2. I don't think it needs persistence. If the collect element is used that way - the orientation is flipped probably flipped many times during entering/communication and it doesn't matter where it starts (or if it's always starting in the default direction which is set in collect element settings)